### PR TITLE
replace `Neg` example with something more evocative of negation

### PR DESCRIPTION
**Pull request by matthew-piziak**

### CODE REVIEW

File: src/libcore/ops.rs [hunk 1]
```diff
@@ -470,26 +470,37 @@ rem_impl_float! { f32 f64 }
 ///
 /// # Examples
 ///
-/// A trivial implementation of `Neg`. When `-Foo` happens, it ends up calling
-/// `neg`, and therefore, `main` prints `Negating!`.
+/// An implementation of `Neg` for `Sign`, which allows the use of `-` to
```
Comment (mrhota) [inline src/libcore/ops.rs]: comma might be incorrect here

Comment (matthew-piziak) [inline src/libcore/ops.rs]: I don't believe so. It's a non-restrictive which clause.

Comment (mrhota) [inline src/libcore/ops.rs]: Correction: the comma might be unnecessary, not strictly incorrect.

Comment (matthew-piziak) [inline src/libcore/ops.rs]: I like how relevant your avatar is for this PR, given that it's a cyclic set of invertible elements. :P

Comment (matthew-piziak) [inline src/libcore/ops.rs]: I stand by the necessity of this comma (Strunk & White II.3), but I certainly won't fight to the death over it.

**Comment by mrhota** [comment]
These examples are far clearer. Thanks

File: src/libcore/ops.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libcore/ops.rs]: I would put the comma here, but I also tend to use commas a lot more than most people.

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thanks!

File: src/libcore/ops.rs (hunk 1 above)
Comment (mrhota) [inline src/libcore/ops.rs]: no no, you have convinced me. not that you needed to :smile:

**Pull request merged**